packages/components/src/disabled/index.tsx

```diff
@@ -0,0 +1,82 @@
+/**
+ * External dependencies
+ */
+import classnames from 'classnames';
+import type { RefCallback } from 'react';
+
+/**
+ * WordPress dependencies
+ */
+import { useDisabled } from '@wordpress/compose';
+import { createContext } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import { StyledWrapper } from './styles/disabled-styles';
+import type { DisabledProps } from './types';
+import type { WordPressComponentProps } from '../ui/context';
+
+const Context = createContext< boolean >( false );
+const { Consumer, Provider } = Context;
+
+/**
+ * `Disabled` is a component which disables descendant tabbable elements and prevents pointer interaction.
+ *
+ * @example
```

Comment:
The `@example` JSDoc keyword is not recognized by StoryBook's docgen, so it should be removed.
 
 See https://github.com/WordPress/gutenberg/blob/trunk/packages/components/CONTRIBUTING.md#refactoring-a-component-to-typescript point 8.

```suggestion
```